impl/client/src/main/javascript/web/pentaho/type/property.js

```diff
@@ -280,6 +281,47 @@ define([
         },
         // endregion
 
+        // region alias attribute
+        _nameAlias: undefined,
+
+        /**
+         * Gets or sets the alias of the _property type_.
+         *
+         * The alias of a _property type_ is an alternative identifier for serialization purposes.
+         *
+         * ### Set
+         *
+         * This attribute must be set when defining a new _property type_,
+         * and cannot be changed afterwards.
+         *
+         * When set to a non-{@link Nully} and non-{@link String} value,
+         * the value is first replaced by the result of calling its `toString` method.
+         *
+         * @type {!nonEmptyString}
+         *
+         * @throws {pentaho.lang.ArgumentRequiredError} When set to an empty string or a _nully_ value.
+         * @throws {TypeError} When set to a value different from the current one.
+         *
+         * @see pentaho.type.spec.IPropertyTypeProto#nameAlias
+         */
+        get nameAlias() {
+          return this._nameAlias;
+        },
+
+        set nameAlias(value) {
+          value = nonEmptyString(value);
+
+          if(!value) throw error.argRequired("nameAlias");
+
+          // Hierarchy consistency
+          var baseValue = this._nameAlias;
+          if(baseValue && value !== baseValue)
```

Comment:
But it can change it if undefined in the base? Can it can raise any inconsistency?